aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -64,6 +64,45 @@ static constexpr int launch_bound2 = 4;
 
 namespace at { namespace native {
 
+// See [NOTE: Complex Operator Unification]
+// std::complex and thrust::complex don't work with some !needs_dynamic_casting optimizations.
+// They always currently map to !needs_dynamic_casting even though we sometimes rely on the ability
+// to reinterpret_cast between these representations.
+// In order to separate these concerns, we have a check for non-c10 complex separately.
+template<typename func_t, int nargs=function_traits<func_t>::arity>
+struct uses_non_c10_complex {
+  constexpr static bool check() {
+    using traits = function_traits<func_t>;
+    using type = typename traits::template arg<nargs - 1>::type;
+    constexpr bool non_c10_complex =
```

Comment:
would it make sense to have `is_non_c10_complex_t` in complex_utils.h, similar to how we have `is_complex_t`?

Reply:
if someone needs it we can certainly move it.

My take is we should just have c10_type like we have ztype and map everything that way.